GNUmakefile

```diff
@@ -280,17 +277,6 @@ test-nomad: dev ## Run Nomad test suites
 		bash -C "$(PROJECT_ROOT)/scripts/test_check.sh" ; \
 	fi
 
-.PHONY: e2e-test
-e2e-test: dev ## Run the Nomad e2e test suite
```

Comment:
There are some CircleCI configurations that call this `make` target (and which are effectively no-op anyways because `NOMAD_E2E` isn't set!). We should remove those as well.

Reply:
Oh! Actually I think it was being run! https://circleci.com/gh/hashicorp/nomad/30307

And we have at least 2 other test packages in `e2e/` that use an `-integration` flag:

1. https://github.com/hashicorp/nomad/blob/v0.10.2/e2e/rescheduling/server_side_restarts_suite_test.go#L11
1. https://github.com/hashicorp/nomad/blob/v0.10.2/e2e/migrations/migrations_test.go#L17

However neither of those are referenced by the e2e-test make target.

Do you have any ideas/opinions on how to sort this out? If not I'll keep digging.

Reply:
> Do you have any ideas/opinions on how to sort this out? If not I'll keep digging.

I have cards on the E2E project board to figure out what's up with those two tests. Maybe this is pointing to a need for a separate set of integration tests that can run in standalone environments rather than needing a whole cluster (although I'm not convinced migrations is one of those that can avoid it). But that's adding a lot of scope to this PR!

Let's:
* keep this running on CircleCI as-is for the time being.
* document in the README for the test that it's not being run under nightly E2E but for every commit in Circle.
* come back to how/where we want to run this once we've has a discussion about whether there's a separate integration test suite we want to run, where that might live in the code base, etc.